cmake/iotjs.cmake

```diff
@@ -566,24 +566,24 @@ endif()
 if(NOT BUILD_LIB_ONLY)
   set(TARGET_IOTJS iotjs)
   message(STATUS "CMAKE_BINARY_DIR        ${CMAKE_BINARY_DIR}")
-  message(STATUS "BINARY_INSTALL_DIR      ${INSTALL_PREFIX}/bin")
-  message(STATUS "LIBRARY_INSTALL_DIR     ${INSTALL_PREFIX}/lib")
+  message(STATUS "BINARY_INSTALL_DIR      ${INSTALL_PREFIX}/${BIN_INSTALL_DIR}")
```

Comment:
Either this message is incorrectly stating that the binary install directory is `${INSTALL_PREFIX}/${BIN_INSTALL_DIR}` or lines 580 and 585 miss the `${INSTALL_PREFIX}/`. Which one? (If the prefix is not needed for "bin", why is that, i.e., why is it still needed for libraries?)

Reply:
yes this is intentional check the updated commit message:

```
Install directory is prefixed by destdir to install in staging directory.

This change is useful for debian packaging.

Runtime destination rely on RUNTIME_OUTPUT_DIRECTORY property

This was introduced first in jerryscript-project#1134 and then droped in jerryscript-project#1848

This will be helpful for packaging,
or example on Tizen install using RPM macro: "%make_install"
of Debian's debhelper.

Work along CMAKE_INSTALL_PREFIX,
or/and can be overloaded individualy to alternate locations.

Email has been updated for community support.

Forwarded: jerryscript-project#1923
Bug: jerryscript-project#1945
Relate-to: jerryscript-project#1134
Change-Id: Ib969a1d5c2e7bc402b455377fc57a94654aa74dc
Origin: https://github.com/TizenTeam/iotjs/tree/sandbox/rzr/review/install/master
Bug-Debian: https://bugs.debian.org/957364
Update: 2020-07-24
IoT.js-DCO-1.0-Signed-off-by: Philippe Coval rzr@users.sf.net
```

I just verified this patch with other debian packaging patches on current base:

https://github.com/jerryscript-project/iotjs/compare/master...TizenTeam:sandbox/rzr/next/snapshot/master?expand=1

Reply:
OK, I've reviewed this commit again. And I still cannot grasp what `${INSTALL_PREFIX}` is for. I've updated my local copy of iotjs and searched for `INSTALL_PREFIX`. All I could find was either `CMAKE_INSTALL_PREFIX` (usually in jerryscript-related files) or `INSTALL_PREFIX`, but this latter one only in this file.

My problem is:
- the messages print all install directories prefixed with `${INSTALL_PREFIX}/`
- `install()` sets all destinations without that prefix.

It may happen that `INSTALL_PREFIX` is not set by anyone at all and is only a superfluous path element? But still, the current master is at least consistent in the sense that it prints the expression that it uses for installing. On the contrary, the commit keeps `INSTALL_PREFIX` in the messages but drops them from install destinations.

If `INSTALL_PREFIX` is unused, it should be dropped from the messages, too, IMO.